llvm/include/llvm/TextAPI/Record.h

```diff
@@ -103,6 +103,10 @@ class GlobalRecord : public Record {
 
   bool isFunction() const { return GV == Kind::Function; }
   bool isVariable() const { return GV == Kind::Variable; }
+  void setKind(const Kind &V) {
```

Comment:
In this case I would prefer to pass the argument by value. It is only an `uint8_t` enum. There is no need to pass it by reference.

Reply:
While I agree, I also wanted to keep the `const` mostly to ensure that the passed value isn't modified, just assigned. But that puts us back to https://github.com/llvm/llvm-project/pull/75006#discussion_r1422148304